### PR TITLE
usb_dc_stm32: fix init on soft reset

### PR DESCRIPTION
**Pull request by dayjaby**
Soft resetting a STM32 device currently does not reset the USB
connection, which causes a few problems: The endpoints do not respond
anymore, and within zephyr any kind of status information like
CDC_SET_CONTROL_LINE_STATE is also missing as they are not
re-negotiated.

This is fixed by stopping the USB device from zephyr side, right after
initialising the USB device.

Tested on a custom board (Olimexino STM32F3 - looking forward to PR this as well):

```
cd samples/subsys/usb/cdc_acm
west build -p auto -b olimexino_stm32f303rc .
arm-none-eabi-gdb build/zephyr/zephyr.elf
(gdb) load
(gdb) mon reset
```

The cdc_acm sample application could be tested in another window:
```
# is necessary in my case to set the DTR:
stty -F /dev/ttyACM1 -hupcl
screen /dev/ttyACM1 921600 8N1
<typed stuff gets echoed here correctly>
```

The logger output with `CONFIG_USB_CDC_ACM_LOG_LEVEL_DBG=y` and `CONFIG_USB_DRIVER_LOG_LEVEL_DBG=y` is attached.
[usb_reset.log](https://github.com/zephyrproject-rtos/zephyr/files/9225986/usb_reset.log)

Signed-off-by: David Jablonski <dayjaby@gmail.com>

### CODE REVIEW

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -453,6 +453,13 @@ static int usb_dc_stm32_init(void)
 		return -EIO;
 	}
 
+	LOG_DBG("HAL_PCD_Stop");
```
Comment (erwango) [inline drivers/usb/device/usb_dc_stm32.c:456]: Please add a quick comment so we can see quickly what was the reason stop is required before start.

**Review by erwango — CHANGES_REQUESTED** [review]
Minor comment otherwise LGTM, thanks!

File: drivers/usb/device/usb_dc_stm32.c (hunk 1 above)
Comment (dayjaby) [reply drivers/usb/device/usb_dc_stm32.c:459]: Done, thanks for your review :)

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for the (quick) update, but please squash the new commit

**Review by erwango — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**